test/framework/spec_helpers.go

```diff
@@ -95,8 +100,86 @@ func DumpAllResourcesAndLogs(ctx context.Context, clusterProxy ClusterProxy, art
+		ShowOtherConditions:     "all",
+		ShowClusterResourceSets: false,
+		ShowTemplates:           false,
+		ShowMachineSets:         true,
+		AddTemplateVirtualNode:  true,
+		Echo:                    true,
+		Grouping:                false,
+		V1Beta2:                 true,
+	})
+	Expect(err).ToNot(HaveOccurred(), "Failed to run clusterctl describe")
+
+	Expect(os.MkdirAll(input.LogFolder, 0750)).To(Succeed(), "Failed to create log folder %s", input.LogFolder)
+
+	f, err := os.Create(filepath.Join(input.LogFolder, fmt.Sprintf("clusterctl-describe-%s.txt", input.Name)))
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a file for saving clusterctl describe output")
+
+	defer f.Close()
+
+	w := bufio.NewWriter(f)
+	cmdtree.PrintObjectTreeV1Beta2(tree, w)
+	Expect(w.Flush()).To(Succeed(), "Failed to save clusterctl describe output")
+}
+
+type DescribeAllClusterInput struct {
+	Client               client.Client
+	KubeConfigPath       string
+	ClusterctlConfigPath string
+	ArtifactFolder       string
+	Namespace            string
+}
+
+func DescribeAllCluster(ctx context.Context, input DescribeAllClusterInput) {
+	clusters := &clusterv1.ClusterList{}
+	Eventually(func() error {
+		return input.Client.List(ctx, clusters, client.InNamespace(input.Namespace))
+	}, retryableOperationTimeout, retryableOperationInterval).Should(Succeed(), "Failed to list clusters in namespace %s", input.Namespace)
+
+	for _, c := range clusters.Items {
+		DescribeCluster(ctx, DescribeClusterInput{
+			ClusterctlConfigPath: input.ClusterctlConfigPath,
+			LogFolder:            filepath.Join(input.ArtifactFolder, "clusters", c.Name),
+			KubeConfigPath:       input.KubeConfigPath,
+			Namespace:            c.Namespace,
+			Name:                 c.Name,
+		})
+	}
+}
+
 // DumpSpecResourcesAndCleanup dumps all the resources in the spec namespace and cleans up the spec namespace.
-func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterProxy ClusterProxy, artifactFolder string, namespace *corev1.Namespace, cancelWatches context.CancelFunc, cluster *clusterv1.Cluster, intervalsGetter func(spec, key string) []interface{}, skipCleanup bool) {
+func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterProxy ClusterProxy, clusterctlConfigPath, artifactFolder string, namespace *corev1.Namespace, cancelWatches context.CancelFunc, cluster *clusterv1.Cluster, intervalsGetter func(spec, key string) []interface{}, skipCleanup bool) {
+	// Describe all clusters
+	DescribeAllCluster(ctx, DescribeAllClusterInput{
+		Client:               clusterProxy.GetClient(),
+		KubeConfigPath:       clusterProxy.GetKubeconfigPath(),
+		ClusterctlConfigPath: clusterctlConfigPath,
+		ArtifactFolder:       artifactFolder,
+		Namespace:            namespace.Name,
+	})
+
 	// Dump all the resources in the spec namespace and the workload cluster.
 	DumpAllResourcesAndLogs(ctx, clusterProxy, artifactFolder, namespace, cluster)
```

Comment:
Please check the other places where we use DumpAllResourcesAndLogs. I think in most (all?) it would make sense to also call DescribeAllCluster or DescribeCluster

Additionally please check the usages of DumpAllResources (apart from DumpAllResourcesAndLogs) and also consider adding DescribeAllCluster or DescribeCluster there.

I think then we use it roughly everywhere :)